pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -312,13 +312,10 @@ func newRevision(set *apps.StatefulSet, revision int64, collisionCount *int32) (
 	if err != nil {
 		return nil, err
 	}
-	selector, err := metav1.LabelSelectorAsSelector(set.Spec.Selector)
-	if err != nil {
-		return nil, err
-	}
+	podLabels := set.Spec.Template.Labels
 	cr, err := history.NewControllerRevision(set,
 		controllerKind,
-		selector,
+		podLabels,
```

Comment:
You can directly put `set.Spec.Template.Labels` here.

Reply:
@enisoc Updated! :)